app/controllers/works_controller.rb

```diff
@@ -0,0 +1,109 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy, :upvote]
+  before_action :current_user, only: [:upvote]
+
+  def order_list(category)
+    return Work.where(category: category).order(votes_count: :desc, created_at: :asc)
+  end
+
+  def index
+    @works = Work.all
+
+    @albums = order_list("album")
+    @books = order_list("book")
+    @movies = order_list("movie")
+  end
+
+  def show
+    if @work.nil?
+      flash[:warning] = "Work not found"
+      redirect_to works_path and return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(works_params)
+
+    if @work.save
+      flash[:success] = "Work added to the ranker"
+      redirect_to work_path(@work) and return
+    else
+      errors = {}
+      @work.errors.each do |column, message|
+        errors[column] = message
+      end
+      flash[:warning] = errors
```

Comment:
When rendering you should use `flash.now` instead of the regular `flash`.  

This will cause the banner to show on the next request as well.